### PR TITLE
We only document use of HS256, no need for others

### PR DESCRIPTION
**Pull request by kumar303**

### CODE REVIEW

File: webpay/api/tests/test_api.py [hunk 1]
```diff
@@ -132,6 +133,12 @@ def test_configures_transaction_fail(self):
         res = self.post(req='')  # cause a form error.
         eq_(res.status_code, 400)
 
+    def test_unsupported_jwt_algorithm(self):
+        with self.settings(SUPPORTED_JWT_ALGORITHMS=['HS256']):
+            res = self.post(
+                request_kwargs={'jwt_kwargs': {'algorithm': 'none'}})
+        eq_(res.status_code, 400)
```
Comment (mstriemer) [inline webpay/api/tests/test_api.py]: Do we give any more feedback than this? Just a `400` seems pretty tough to debug.

File: webpay/api/tests/test_api.py [hunk 2]
```diff
@@ -132,6 +133,12 @@ def test_configures_transaction_fail(self):
         res = self.post(req='')  # cause a form error.
         eq_(res.status_code, 400)
 
+    def test_unsupported_jwt_algorithm(self):
+        with self.settings(SUPPORTED_JWT_ALGORITHMS=['HS256']):
+            res = self.post(
+                request_kwargs={'jwt_kwargs': {'algorithm': 'none'}})
```
Comment (mstriemer) [inline webpay/api/tests/test_api.py]: The code this is testing seems to check `'alg'` not `'algorithm'`.

File: webpay/pay/forms.py [hunk 3]
```diff
@@ -56,6 +58,15 @@ def clean_req(self):
             log.info('JWT was not a dict, it was %r' % type(payload))
             raise forms.ValidationError(msg.INVALID_JWT_OBJ)
 
+        # The decode method already parses the header like this
+        # but does not give us access to it so we need to re-parse it.
+        header = json.loads(jwt.base64url_decode(jwt_data.split('.')[0]))
+        if header['alg'] not in settings.SUPPORTED_JWT_ALGORITHMS:
```
Comment (mstriemer) [inline webpay/pay/forms.py]: Unless this is something else? Should this be `header['algorithm']`?

File: webpay/api/tests/test_api.py (hunk 2 above)
Comment (kumar303) [inline webpay/api/tests/test_api.py]: ``{algorithm: none}`` is just a keyword sent to the JWT library which produces a JWT with the standard ``alg`` field. I did see the test fail before the patch so it should be working.

**Comment by mstriemer** [comment]
It would be nice to get a better error than just `400` but that isn't the end of the world.

r+

File: webpay/api/tests/test_api.py (hunk 1 above)
Comment (kumar303) [inline webpay/api/tests/test_api.py]: oh yeah, I forgot this is a standard JSON API response so we have a readable error code. Thanks.

**Pull request merged**